include/swift/Basic/ThreadSafeRefCounted.h

```diff
@@ -28,18 +28,19 @@ namespace swift {
 /// FIXME: This should eventually move to llvm.
 template <class Derived>
 class ThreadSafeRefCountedBase {
-  mutable llvm::sys::cas_flag ref_cnt;
+  mutable std::atomic<unsigned> ref_cnt;
 
 protected:
   ThreadSafeRefCountedBase() : ref_cnt(0) {}
 
 public:
   void Retain() const {
-    llvm::sys::AtomicIncrement(&ref_cnt);
+    ref_cnt.fetch_add(1, std::memory_order_relaxed);
   }
 
   void Release() const {
-    int refCount = static_cast<int>(llvm::sys::AtomicDecrement(&ref_cnt));
+    int refCount =
+        static_cast<int>(ref_cnt.fetch_sub(1, std::memory_order_relaxed));
```

Comment:
Yes, that seems safer.  However, this is maintaining the status quo (the current behaviour of AtomicDecrement is a memory_order_relaxed).  I can do a subsequent change to change it to memory_order_acquire, or we can make it part of this change.  I think doing a follow up is better as this change doesnt change semantics that way.